File: Dev/C++/Mascaret/src/HAVE/EmbodiedAgent.cpp
#include "HAVE/EmbodiedAgent.h"
#include "HAVE/Body.h"

namespace HAVE {

EmbodiedAgent::EmbodiedAgent(shared_ptr<AgentPlateform> ap, const string & name, shared_ptr<AgentClass> agentclass) : Agent(ap, name, agentclass)
{
   _body = shared_ptr<Body> (new Body(this));
}

EmbodiedAgent::~EmbodiedAgent()
{}

}

File: Dev/CS/Mascaret/Mascaret/HAVE/EmbodiedAgent.cs
using System;
using System.Collections.Generic;


namespace Mascaret
{
    public class EmbodiedAgent : Agent
    {
        public BehaviorRealizer behaviorRealizer;

        private Body body;
        public Body Body
        {
            get { return body; }
            set { body = value; }
        }

        public EmbodiedAgent(AgentPlateform ap, string name, AgentClass agentclass)
            : base(ap, name, agentclass, "default")
        {
            body = new Body(this);
        }

        public override void addIntention(string fml)
        {
            List<string> bmlList = behaviorPlanner.parseIntention(fml);
            foreach (string bmlI in bmlList)
                this.behaviorRealizer.addBehavior(bmlI);
        }

    }
}


